app/src/main/kotlin/com/wire/android/notification/CallNotificationManager.kt

```diff
@@ -54,6 +55,12 @@ class CallNotificationManager @Inject constructor(private val context: Context)
 
     fun hideIncomingCallNotification() {
         appLogger.i("$TAG: hiding incoming call")
+
+        // This delay is just so when the user receives two calling signals one straight after the other [INCOMING -> CANCEL]
+        // Due to the signals being one after the other we are creating a notification when we are trying to cancel it, it wasn't properly
+        // cancelling vibration as probably when we were cancelling, the vibration object was still being created and started and thus
+        // never stopped.
+        TimeUnit.MILLISECONDS.sleep(CANCEL_CALL_NOTIFICATION_DELAY)
```

Comment:
Sleep is putting a Thread to sleep right ? Perhaps we can do it on coroutine level and like suspend the coroutine instead of putting a Thread to sleep ?